core/src/components/content/content.tsx

```diff
@@ -307,6 +307,7 @@ export class Content implements ComponentInterface {
     const { scrollX, scrollY } = this;
     const mode = getIonMode(this);
     const forceOverscroll = this.shouldForceOverscroll();
+    const TagType = this.el.closest('ion-menu, ion-popover, ion-modal') !== null ? 'div' : 'main' as any;
```

Comment:
Two things here:

1. This probably doesn't need to be reactive and might result in a small performance hit. We can just do this query selector in `connectedCallback` and cache the result:

```tsx
private isMainContent = true;

...

connectedCallback() {
  this.isMainContent = this.el.closest('ion-menu, ion-popover, ion-modal') === null;
}

...

const { isMainContent } = this;

const TagType = isMainContent ? 'main' : 'div';
```

`connectedCallback` is called when `ion-content` is added to the DOM.

2. If possible, try to avoid using the `any` type.